Editorial review: Add docs for scroll-target-group (#40843)
mdn/content: files/en-us/web/css/scroll-target-group/index.md

---
title: scroll-target-group
slug: Web/CSS/scroll-target-group
page-type: css-property
browser-compat: css.properties.scroll-target-group
sidebar: cssref
---

The **`scroll-target-group`** [CSS](/en-US/docs/Web/CSS) property specifies whether a container element is a scroll marker group container.

## Syntax

```css
/* Keyword values */
scroll-target-group: auto;
scroll-target-group: none;

/* Global values */
scroll-target-group: inherit;
scroll-target-group: initial;
scroll-target-group: revert;
scroll-target-group: revert-layer;
scroll-target-group: unset;
```

The `scroll-target-group` property is specified as one of the keyword values listed below.

> ### Values

Reviewer: Thank you!

- `auto`
  - : The element is a scroll marker group container.
- `none`
  - : The element is not a scroll marker group container.

## Description

> Setting `scroll-target-group: auto` on a container element denotes it as a **scroll marker group container**, which groups together a set of navigation items that allow you to navigate between elements on a page (such as carousel items or article sections) and are highlighted to indicate which element is currently in view.

Reviewer: should the "you" here be "users" (that is, not the devs)?
```suggestion
Setting `scroll-target-group: auto` on an element denotes it as a **scroll marker group container**. This groups together a set of navigation items that allow you to navigate between elements on a page (such as carousel items or article sections) and highlight which element is currently in view.
```

Any {{htmlelement("a")}} elements with fragment identifiers inside the container are automatically set as scroll markers. The anchor element whose scroll target is currently in view can be styled via the {{cssxref(":target-current")}} pseudo-class.

> Scroll marker group containers created using `scroll-target-group` behave in a very similar way to those created using the {{cssxref("scroll-marker-group")}} property, with some differences:

Reviewer: We're listing the differences but the points explain what `scroll-marker-group` does without spelling out how that's different from `scroll-target-group`. Also since this page is about `scroll-target-group`, maybe we should first state what happens with `scroll-target-group` and then contrast it with `scroll-marker-group`.

Reviewer: Also, how about an h3 for differences between `scroll-target-group` and `scroll-marker-group` so then it can be linked from `scroll-marker-group` page?

> - `scroll-marker-group` automatically creates a set of pseudo-elements to represent the group container ({{cssxref("::scroll-marker-group")}}) and markers (one or more instances of {{cssxref("::scroll-marker")}}), and they automatically have the expected navigation associations made with the {{glossary("scroll container")}} the group container is generated on. This is quicker to set up, and you don't need to create markup for the container and markers, but creating your own markup and setting it as a scroll marker group container via `scroll-target-group` provides more control and flexibility.

Reviewer: ```suggestion
- `scroll-marker-group` automatically creates a set of pseudo-elements to represent the group container ({{cssxref("::scroll-marker-group")}}) and the markers (one or more instances of {{cssxref("::scroll-marker")}}). These automatically have the expected navigation associations with the {{glossary("scroll container")}} they are generated on. This is quicker to set up because you don't need to create markup for the container and markers. However, creating your own markup and setting it as a scroll marker group container via `scroll-target-group` gives you more control and flexibility.
```

> - A group container generated with `scroll-marker-group` automatically has [`tablist`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tablist_role)/[`tab`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tab_role) semantics applied to it, meaning that it behaves like a single item in the tab index, and you can move between scroll markers using cursor keys. The markers specified using `scroll-target-group` behave like individual links by default, but you have the flexibility of providing alternative semantics and behavior if wished.

Reviewer: should the "you" be "users"?
```suggestion
- A group container generated with `scroll-marker-group` automatically has [`tablist`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tablist_role)/[`tab`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tab_role) semantics applied to it, meaning that it behaves like a single item in the tab order, and you can move between scroll markers with the arrow keys. The markers specified using `scroll-target-group` behave like individual links by default, but you have the flexibility of providing alternative semantics and behavior.
```

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Basic `scroll-target-group` usage

> This example provides a basic page of content with a table of contents to link to different sections.

Reviewer: or simply
```suggestion
This example shows a page with a table of contents linking to different sections.
```

    narwhal asymmetrical. Af health goth shaman in slow-carb godard echo park.
    Tofu farm-to-table labore salvia tote bag food truck dolore gluten-free
    poutine kombucha fanny pack +1 franzen lyft fugiat. Chicharrones next level
    jianbing, enamel pin seitan cardigan bruh snackwave beard incididunt dolor
    lumbersexual before they sold out dreamcatcher single-origin coffee.
  </p>
</section>
<section id="ch4" class="chapter">
  <h2>Chapter 4</h2>
  <p>
    Austin mukbang scenester pabst, kale chips helvetica in selvage tote bag
    drinking vinegar craft beer pickled meh subway tile +1. Big mood kogi blog,
    vape hella seitan veniam.
  </p>
  <p>
    I'm baby xOXO bespoke cupidatat PBR&B, affogato cronut 3 wolf moon ea
    narwhal asymmetrical. Af health goth shaman in slow-carb godard echo park.
    Tofu farm-to-table labore salvia tote bag food truck dolore gluten-free
    poutine kombucha fanny pack +1 franzen lyft fugiat. Chicharrones next level
    jianbing, enamel pin seitan cardigan bruh snackwave beard incididunt dolor
    lumbersexual before they sold out dreamcatcher single-origin coffee.
  </p>
</section>
```

#### CSS

> We've hidden most of the styling for brevity. Most pertinently to this example, we've set `scroll-target-group: auto` on the `<ol>` to turn it into a scroll marker group container and trigger the browser's algorithm for calculating which `<a>` element is the `:target-current` at any moment in time (that is, which link's target is currently in view). We then style the `:target-current` pseudo-class with a {{cssxref("color")}} of red so that it is clearly visible.

Reviewer: ```suggestion
We've hidden most of the styling for brevity. Most pertinent to this example, we've set `scroll-target-group: auto` on the `<ol>` to turn it into a scroll marker group container and trigger the browser's algorithm for calculating which `<a>` element is the `:target-current` at any given time (that is, which link's target is currently in view). We then style the `:target-current` pseudo-class with a `red` {{cssxref("color")}} so that it stands out clearly.
```

```css hidden live-sample___basic-usage
body {
  font: 1.2em / 1.5 system-ui;
  width: 50%;
  max-width: 700px;
  margin: 0 auto;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
}

section {
  padding-top: 60px;
}

a {
  color: black;
  text-decoration: none;
}

a:hover,
a:focus {
  text-decoration: underline;
}

ol {
  display: flex;
  width: 100%;
  justify-content: space-around;
  list-style-type: none;
  padding: 20px 0;
  margin: 0;
  background: white;
}
```

```css live-sample___basic-usage
ol {
  scroll-target-group: auto;
}

:target-current {
  color: red;
}
```

#### Result

Try navigating by activating the links and by scrolling. You'll see that in each case, the red highlight moves between the links to match the section currently being shown.

> {{ EmbedLiveSample("basic-usage", "100%", 500) }}

Reviewer: ```suggestion
{{EmbedLiveSample("basic-usage", "100%", 500)}}
```

}

ul::scroll-button(*):active {
  translate: 1px 1px;
}

ul::scroll-button(*):disabled {
  color: rgb(0 0 0 / 0.2);
}

ul::scroll-button(left) {
  content: "◄";
}

ul::scroll-button(right) {
  content: "►";
}

/* Position the scroll buttons */

ul {
  anchor-name: --myCarousel;
}

ul::scroll-button(*) {
  position: absolute;
  position-anchor: --myCarousel;
}

ul::scroll-button(left) {
  right: calc(anchor(left) - 70px);
  top: calc(anchor(bottom) - 80px);
}

ul::scroll-button(right) {
  left: calc(anchor(right) - 70px);
  top: calc(anchor(bottom) - 80px);
}
```

```css live-sample___carousel
ol {
  position: absolute;
  position-anchor: --myCarousel;
  top: calc(anchor(bottom) - 90px);
  justify-self: anchor-center;
  display: flex;
  justify-content: center;
  gap: 20px;

  list-style-type: none;
  padding: 0;
  scroll-target-group: auto;
}

ol a {
  width: 28px;
  height: 28px;
  display: inline-block;
  text-align: center;
  text-decoration: none;
  padding-top: 4px;
  color: black;
  background-color: transparent;
  border: 2px solid black;
  border-radius: 50%;
}

ol a:hover,
ol a:focus,
:target-current {
  background-color: black;
  color: white;
}
```

#### Result

> Try navigating by activating the scroll marker links, scrolling, and pressing the scroll buttons. You'll see that in each case, the highlight moves between the links to match the section currently being shown.

Reviewer: we can probably clarify that there are three ways to navigate
```suggestion
Try navigating in of these three ways: by activating the scroll marker links, by horizontal scrolling, or by pressing the scroll buttons on either side. You'll see that in each case, the highlight moves between the links to match the section currently being shown.
```
